climate_index_collection/indices.py

```diff
@@ -79,7 +81,9 @@ def southern_annular_mode_pc(data_set, geopoth_name="geopotential-height"):
 
     mask = data_set.coords["lat"] <= -20
 
-    geopoth = data_set[geopoth_name].where(mask)
+    geopoth = data_set[geopoth_name]
+    geopoth = geopoth * np.sqrt(np.cos(np.deg2rad(geopoth.lat)))
```

Comment:
I like this much simpler approach.

Still, I'd argue for indirecting the weight calculation to a function which can be reused in an easier way:

```python
def eof_weights(dobj):
    return np.sqrt(np.cos(np.deg2rad(dobj.coords["lat"])))
```
and then use, e.g., `geopoth = geopoth * eof_weights(geopoth.)`

Reply:
Weight calculation is now done in a separate function, as you proposed.